lib/src/directory_watcher.dart

```diff
@@ -29,6 +30,14 @@ abstract class DirectoryWatcher implements Watcher {
   /// watchers.
   factory DirectoryWatcher(String directory, {Duration pollingDelay}) {
     if (FileSystemEntity.isWatchSupported) {
+      for (var custom in customWatcherFactories) {
```

Comment:
Optional: I suggest putting this logic (and for files) in custom_watcher_factory.dart so it's with the description of the behaviour.